### ISSUE TITLE
AWSManagedMachinePool IAM Role too long

### ISSUE THREAD

**Issue by cabrinha** [kubernetes-sigs/cluster-api-provider-aws#2543]
/kind bug

**What steps did you take and what happened:**
Created an AWSManagedMachinePool and the IAM Role that was automatically generated had a name that was too long.

**What did you expect to happen:**
IAM Role should be the correct length

**Anything else you would like to add:**
```
    - lastTransitionTime: "2021-06-19T02:20:59Z"
      message: "ValidationError: 1 validation error detected: Value 'us-west-2-eng-mgmt-us-west-2-eng-mgmt-general-nodegroup-iam-service-role'
        at 'roleName' failed to satisfy constraint: Member must have length less than
        or equal to 64\n\tstatus code: 400, request id: ed6c3cd0-9dfc-43d7-8c7a-342792ff4c3f"
```

**Environment:**

- Cluster-api-provider-aws version: latest
- Kubernetes version: (use `kubectl version`): v1.19
- OS (e.g. from `/etc/os-release`): Amazon Linux 2

**Comment by richardcase** [comment]
Thanks @cabrinha 

/area provider/eks
/milestone v0.7.0
/help
/priority important-soon
/good-first-issue

**Comment by richardcase** [comment]
Fixed by #2559
